src/test/java/com/google/neighborgood/TaskServletTest.java

```diff
@@ -0,0 +1,172 @@
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableMap;
+import java.io.*;
+import javax.servlet.http.*;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit test on the TaskServlet file */
+@RunWith(JUnit4.class)
+public final class TaskServletTest {
+
+  /* Set up the test environment with Datastore and UserService and simulate the situation
+   * that the user is logged in as admin, has a specific email and userID.
+   */
+  private LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true)
+          .setEnvIsLoggedIn(true)
+          .setEnvEmail("leonardzhang@google.com")
+          .setEnvAuthDomain("1234567890")
+          .setEnvAttributes(
+              ImmutableMap.of(
+                  "com.google.appengine.api.users.UserService.user_id_key", "1234567890"));
+
+  private UserService userService;
+  private DatastoreService ds;
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    userService = UserServiceFactory.getUserService();
+    ds = DatastoreServiceFactory.getDatastoreService();
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void setUpTest() {
+    // Test the UserService feature
+    assertTrue(userService.isUserAdmin());
+    assertTrue(userService.isUserLoggedIn());
+
+    // Test the DataStore feature
+    assertEquals(0, ds.prepare(new Query("dummy")).countEntities(withLimit(10)));
+    ds.put(new Entity("dummy"));
+    ds.put(new Entity("dummy"));
+    assertEquals(2, ds.prepare(new Query("dummy")).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void singleInputDoPostTest() {
+    // Check whether the datastore is empty before the test
+    assertEquals(0, ds.prepare(new Query("Task")).countEntities(withLimit(10)));
+
+    when(request.getParameter("reward-input")).thenReturn("50");
+    when(request.getParameter("task-detail-input")).thenReturn("Help me please");
+
+    try {
+      new TaskServlet().doPost(request, response);
+    } catch (IOException e) {
+      assertTrue(false);
+    }
+
+    // After sending the POST request, there should be one entity in the datastore
+    assertEquals(1, ds.prepare(new Query("Task")).countEntities(withLimit(10)));
+    PreparedQuery results = ds.prepare(new Query("Task"));
+    Entity entity = results.asSingleEntity();
+    if (entity == null) {
+      // The entity can't be null
+      assertTrue(false);
+    }
```

Comment:
`assertNotNull(entity)` is a more concise way to express this

Reply:
Ah this is way more concise. I will switch to this in the new commit.